src/twisted/internet/_resolver.py

```diff
@@ -26,14 +39,25 @@
 from twisted.internet.defer import Deferred
 from twisted.internet.error import DNSLookupError
 from twisted.internet.interfaces import (
+    IAddress,
     IHostnameResolver,
     IHostResolution,
+    IReactorThreads,
     IResolutionReceiver,
     IResolverSimple,
 )
 from twisted.internet.threads import deferToThreadPool
 from twisted.logger import Logger
 from twisted.python.compat import nativeString
+from twisted.python.failure import Failure
+
+if TYPE_CHECKING:
+    from twisted.python.runtime import platform
+
+    if platform.supportsThreads():
```

Comment:
What does it mean to have an executable condition like this inside an `if TYPE_CHECKING` block, where the code will definitely never run?

Reply:
I think it works fine, I believe anything inside of `TYPE_CHECKING` is just ignored when CPython executes the code (hence why `ThreadPool` is quoted below). mypy seems to be OK with it, but I'm not sure if that actually executes the module (or how it picks up on what this is doing).

Note that this conditional is copied from `twisted.internet.interfaces`:

https://github.com/twisted/twisted/blob/release-21.2.0-10091/src/twisted/internet/interfaces.py#L56-L59